stories-table-add-search

From: Bryan Larsen <bryan@larsen.st>

To get the search feature working, we need to update the controller side. Add a `show` method to `app/controllers/projects_controller.rb` and update the `<table-plus>` to use `@stories` as it's context:

SHOW_PATCH

There's a lot happening here, so we're just going to point you to appropriate manual chapters to explain the concepts.

- [DRYML context switching](/manual/dryml-guide#the_implicit_context)
- [Changing controller action behaviour](/manual/controllers#changing_action_behaviour)
- [search and order_by scopes](/manual/scopes#static_scopes)
- [find_instance](/manual/controller#find_instance)
- [parse_sort_param](/manual/controller#parse_sort_param)
---
 app/controllers/projects_controller.rb |    8 ++++++++
 app/views/projects/show.dryml          |    2 +-
 2 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/app/controllers/projects_controller.rb b/app/controllers/projects_controller.rb
index 210b07d..3c3b39b 100644
--- a/app/controllers/projects_controller.rb
+++ b/app/controllers/projects_controller.rb
@@ -4,4 +4,12 @@ class ProjectsController < ApplicationController
 
   auto_actions :all
 
+  def show
+    @project = find_instance
+    @stories = @project.stories.
+      search(params[:search], :title).
+      order_by(parse_sort_param(:title, :status))
+    hobo_show
+  end
+
 end
diff --git a/app/views/projects/show.dryml b/app/views/projects/show.dryml
index f62b0b2..0ea6a1d 100644
--- a/app/views/projects/show.dryml
+++ b/app/views/projects/show.dryml
@@ -1,6 +1,6 @@
 <show-page>
   <collection: replace>
-  <table-plus:stories fields="this, tasks_count, status">
+  <table-plus with="&@stories" fields="this, tasks_count, status">
     <empty-message:>No stories match your criteria</empty-message:>
   </table-plus>
   </collection:>
